src/views/tables/PositionsTable.tsx

```diff
@@ -360,31 +359,29 @@ export const PositionsTable = ({
   const perpetualMarkets = useSelector(getPerpetualMarkets, shallowEqual) || {};
   const assets = useSelector(getAssets, shallowEqual) || {};
   const openPositions = useSelector(getExistingOpenPositions, shallowEqual) || [];
-  const openOrders = useSelector(getSubaccountOpenOrders, shallowEqual) || [];
-
-  const stopLossOrders: SubaccountOrder[] = [];
-  const takeProfitOrders: SubaccountOrder[] = [];
-
-  openOrders.map((order: SubaccountOrder) => {
-    if (isStopLossOrder(order)) {
-      stopLossOrders.push(order);
-    } else if (isTakeProfitOrder(order)) {
-      takeProfitOrders.push(order);
-    }
-  });
+  const { stopLossOrders: allStopLossOrders, takeProfitOrders: allTakeProfitOrders } = useSelector(
+    getSubaccountConditionalOrders,
+    shallowEqual
+  );
 
   const positionsData = useMemo(
     () =>
-      openPositions.map((position: SubaccountPosition) => ({
-        tickSizeDecimals:
-          perpetualMarkets?.[position.id]?.configs?.tickSizeDecimals || USD_DECIMALS,
-        asset: assets?.[position.assetId],
-        oraclePrice: perpetualMarkets?.[position.id]?.oraclePrice,
-        stopLossOrders: stopLossOrders.filter((order) => order.marketId === position.id),
-        takeProfitOrders: takeProfitOrders.filter((order) => order.marketId === position.id),
-        ...position,
-      })),
-    [openPositions, perpetualMarkets, assets, openOrders]
+      openPositions.map((position: SubaccountPosition) => {
+        return {
+          tickSizeDecimals:
+            perpetualMarkets?.[position.id]?.configs?.tickSizeDecimals || USD_DECIMALS,
+          asset: assets?.[position.assetId],
+          oraclePrice: perpetualMarkets?.[position.id]?.oraclePrice,
+          stopLossOrders: allStopLossOrders.filter(
+            (order: SubaccountOrder) => order.marketId === position.id
+          ),
+          takeProfitOrders: allTakeProfitOrders.filter(
+            (order: SubaccountOrder) => order.marketId === position.id
+          ),
+          ...position,
+        };
+      }),
+    [openPositions, perpetualMarkets, assets, allStopLossOrders, allTakeProfitOrders]
```

Comment:
yay, can we double check the table doesn't flicker in portfolio? might have to try a busy address like `dydx14rplxdyycc6wxmgl8fggppgq4774l70zt6phkw` on staging

Reply:
actually yea right now it's flickering T__T https://v4-staging-git-mulan-ct-627-triggers-button-0f4921-dydxprotocol.vercel.app/portfolio/overview?address=dydx14rplxdyycc6wxmgl8fggppgq4774l70zt6phkw

Reply:
re positions filter bug -- can this be `positions` instead of `openPositions`?